### DISCUSSION TITLE
Implementation for React Server Components for emails

### DISCUSSION THREAD

**Discussion by robin-ln** [resend/react-email#1240] [Ideas]
### Describe the Bug

i cant use rsc components : 
```
import { Head, Html, Preview } from '@react-email/components';
import { getTranslations } from 'next-intl/server';
import { FC } from 'react';

export const TestEmail: FC = async () => {
  const t = await getTranslations('MagicLinkEmail');

  return (
    <Html>
      <Head />
      <Preview>{t('heading')}</Preview>
    </Html>
  );
};

export default TestEmail;
```

i get this error : 
```
Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.
```

### Link to the code that reproduces this issue

https://gitlab.com/robin.louarn/programme-tz-2023

### To Reproduce

```
pnpm install
pnpm email:dev
```

### Expected Behavior

get a preview of my mail

### What's your node version? (if relevant)

20.11.0

**Comment by gabrielmfern** [comment]
Would be nice to be able to, but yes, we don't. This should be mostly a discussion rather than an issue.

**Reply by gvzq** [reply]
Hey @gabrielmfern, what's the solution? This will be an issue for `"react": "^19.0.0-rc-6230622a1a-20240610"` and `"next": "15.0.0-rc.0"`

**Comment by BohdanK-W32** [comment]
It's still not implemented?

**Reply by gvzq** [reply]
@BohdanK-W32 I found server components in the `canary` version. 
 Here is an example: https://github.com/dBianchii/react-email-starter-rsc-bug/pull/1/files

**Reply by BohdanK-W32** [reply]
@gvzq, thanks for letting me know!!

**Comment by Mr-Vipi** [comment]
still not usable with the lastest react-email version

<img width="1046" height="356" alt="image" src="https://github.com/user-attachments/assets/be13686a-c86b-4eea-9475-beda7b9cdec4" />
<img width="478" height="91" alt="image" src="https://github.com/user-attachments/assets/1e5dc84b-88e7-47a4-b169-ab45aa8ed694" />

even though If I render it in server component before sending it works but not able to see it in preview server
